src/Symfony/Component/Validator/Constraints/NotCompromisedPasswordValidator.php

```diff
@@ -44,6 +45,7 @@ public function __construct(HttpClientInterface $httpClient = null, string $char
         $this->httpClient = $httpClient ?? HttpClient::create();
         $this->charset = $charset;
         $this->disabled = $disabled;
+        $this->endpoint = $endpoint;
```

Comment:
it would be useful to allow null and use `$endpoint ?? self::RANGE_API` here

Reply:
Great idea, I'll change it. Tomorrow i'm out of office, so on Friday :).

Reply:
Also, `RANGE_API` was the right name when we only used the Pwned service and one of its APIs. But now that this validator is more generic, you could rename `RANGE_API` as `API_ENDPOINT` (or `DEFAULT_API_ENDPOINT`, etc.)

Reply:
DEFAULT_API_ENDPOINT sounds good, clear and easy.
Maybe we should add a note what the endpoint should do.

BTW, I'm currently working on a simple server implementing the range search in an efficient way (Hash buckets are created using the source data file via a small C program at build-time, then stored in file system for efficient retrieval and caching).